File: src/main/java/com/arnas/klatrebackend/features/invites/GroupInvite.java
package com.arnas.klatrebackend.features.invites;

import org.springframework.lang.NonNull;

public record GroupInvite(
    long id,
    long userId,
    long senderId,
    long groupId,
    @NonNull String status,
    Long acceptedAt,
    Long declinedAt,
    Long revokedAt

) { }

File: src/main/kotlin/com/arnas/klatrebackend/features/invites/GroupInvite.kt
package com.arnas.klatrebackend.features.invites

import com.arnas.klatrebackend.features.groups.Group
import com.arnas.klatrebackend.features.users.User

data class GroupInvite(
    val id: Long,
    val userId: Long,
    val senderId: Long,
    val groupId: Long,
    val status: String,
    val acceptedAt: Long?,
    val declinedAt: Long?,
    val revokedAt: Long?
)

data class GroupInviteDisplay(
    val id: Long,
    val group: Group,
    val sender: User,
    val status: String
)

